jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```diff
@@ -757,11 +757,9 @@ protected ClassLoader enterScope(Request contextRequest)
             catch (Throwable x)
             {
                 // Log as warning the first time it happens.
-                String msg = "Error setting a context classloader on thread {}";
-                if (_enterScopeSetClassloaderFailed.compareAndSet(false, true))
-                    LOG.warn(msg, Thread.currentThread(), x);
-                else if (LOG.isDebugEnabled())
-                    LOG.debug(msg, Thread.currentThread(), x);
+                (_enterScopeSetClassloaderFailed.compareAndSet(false, true) ? LOG.atWarn() : LOG.atDebug())
+                    .setCause(x)
+                    .log("Error setting a context classloader on thread {}", Thread.currentThread());
```

Comment:
@joakime @sbordet Also with the fluent style, code like this is much nicer as it avoids duplicating the log call!